src/lib/encoder.test.ts

```diff
@@ -62,15 +60,12 @@ describe('encoder', () => {
       {
         valueType: 'bytes4',
         decodedValue: '0x13370000',
-        encodedValue:
-          '0x1337000000000000000000000000000000000000000000000000000000000000',
+        encodedValue: '0x13370000',
       },
       {
         valueType: 'bytes',
-        decodedValue:
-          '0x0000000000000000000000000000000000000000000000000000000000001337',
-        encodedValue:
-          '0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000001337',
+        decodedValue: '0x1337',
```

Comment:
Maybe we also want to keep that previous test case:

What if the input is: `0x0000000000000000000000000000000000000000000000000000000000001337` ??

Shall we expect this as an output: `0x1337` ?

We are not limited in tests, we can add more and try to go outside the happy path too.

Reply:
Yeah lets add this test case:

```js
{
    valueType: 'bytes4',
    decodedValue: '0x000000000001337',
    encodedValue: '0x1337',
}
```

Might cause issue on decoding test as it is not symetric:

`0x000000000001337` -> Encode -> `0x1337` -> Decode -> `0x1337`

But we can make it work, i think it is nice to have tests like this. (Apply to other types too)

Reply:
Or... we just throw?

Bc what if:

```
{ valueType: 'bytes4', decodedValue: '0x00000000beef1337', encodedValue: '0x1337', }
```
?

Maybe throwing is safer. As i said in other comment, we don't slice anything. We ask ppl to put the right input

Reply:
I think if the input is bigger than the type, we should throw. This feels safer imo